main.py

```diff
@@ -7,34 +7,60 @@
 
 class Shot(Image):
     velocity = Vector(0, 0)
+    shot_speed = 600
+    def clone(self):
+        clone = Shot()
+        clone.id = self.id
+        clone.source = self.source
+        clone.allow_stretch = self.allow_stretch
+        clone.size = self.size
+        clone.center = self.center
+        return clone
+    
     def update(self, dt):
-        position = Vector(self.center) + self.velocity
+        position = Vector(self.center) + self.velocity * dt
         if (position.x < 0 or position.x > self.parent.width or position.y < 0 or position.y > self.parent.height):
             self.velocity = Vector(0, 0)
             position = Vector(self.parent.width / 2, 0)
         self.center = position
+        
+    def shoot(self, direction):
+        self.velocity = direction * self.shot_speed
 
 class SpawningScreen(Widget):
     shot = ObjectProperty(None)
+    shots = []
```

Comment:
Defining an empty array for the shots and a variable for the maximum number of simultaneous shots.